tsdb/migrate/migrate.go

```diff
@@ -283,6 +299,40 @@ func (m *Migrator) createBucket(db, rp string) (influxdb.ID, error) {
 	return bucket.ID, nil
 }
 
+// Load and extract retention policy from meta.db
+func (m *Migrator) getRetentionPolicy(dbFilter, rpFilter string) (*RetentionPolicyInfo, error) {
+	file := filepath.Join(m.SourcePath, "meta/"+metaFile)
+
+	f, err := os.Open(file)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, err
+		}
+		return nil, err
+	}
+	defer f.Close()
+
+	data, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This can all be shortened to:

```go
data, err := ioutil.ReadFile(file)
if err != nil {
	return nil, err
}
```

Also, your check for `os.IsNotExist(err)` on L308 is redundant since both error cases simply return `err`.